templates/client-snapshot-agent-deployment.yaml

```diff
@@ -0,0 +1,116 @@
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.client.snapshotAgent.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "consul.name" . }}
+      chart: {{ template "consul.chart" . }}
+      release: {{ .Release.Name }}
+      component: client-snapshot-agent
+  template:
+    metadata:
+      labels:
+        app: {{ template "consul.name" . }}
+        chart: {{ template "consul.chart" . }}
+        release: {{ .Release.Name }}
+        component: client-snapshot-agent
+      annotations:
+        "consul.hashicorp.com/connect-inject": "false"
+    spec:
+      {{- if .Values.client.tolerations }}
+      tolerations:
+        {{ tpl .Values.client.tolerations . | nindent 8 | trim }}
+      {{- end }}
+      terminationGracePeriodSeconds: 10
+      serviceAccountName: {{ template "consul.fullname" . }}-client
+
+      {{- if .Values.client.priorityClassName }}
+      priorityClassName: {{ .Values.client.priorityClassName | quote }}
+      {{- end }}
+      {{- if (or .Values.global.bootstrapACLs (and .Values.client.snapshotAgent.configSecret.secretName .Values.client.snapshotAgent.configSecret.secretKey)) }}
+      volumes:
+        {{- if (and .Values.client.snapshotAgent.configSecret.secretName .Values.client.snapshotAgent.configSecret.secretKey) }}
+        - name: snapshot-config
+          secret:
+            secretName: {{ .Values.client.snapshotAgent.configSecret.secretName }}
+            items:
+            - key: {{ .Values.client.snapshotAgent.configSecret.secretKey }}
+              path: snapshot-config.json
+        {{- end }}
+        {{- if .Values.global.bootstrapACLs }}
+        - name: aclconfig
+          emptyDir: {}
+        {{- end }}
+      {{- end }}
+      containers:
+        - name: consul-snapshot-agent
+          image: "{{ default .Values.global.image .Values.client.image }}"
+          env:
+            - name: HOST_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP
+            {{- if .Values.global.bootstrapACLs }}
+            - name: CONSUL_HTTP_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-consul-client-snapshot-agent-acl-token"
+                  key: "token"
+            {{- end}}
+          command:
+            - "/bin/sh"
+            - "-ec"
+            - |
+              exec /bin/consul snapshot agent \
+                {{- if (and .Values.client.snapshotAgent.configSecret.secretName .Values.client.snapshotAgent.configSecret.secretKey) }}
+                -config-dir=/consul/config \
+                {{- end }}
+                {{- if .Values.global.bootstrapACLs}}
+                -config-dir=/consul/aclconfig \
+                {{- end }}
+                -http-addr=http://${HOST_IP}:8500
```

Comment:
ahh, is the `/bin/sh` for `HOST_IP`?

Reply:
although not sure it's necessary https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/#using-environment-variables-inside-of-your-config